tools/gfx/render-d3d12.cpp

```diff
@@ -1413,37 +1513,51 @@ Result D3D12Renderer::initialize(const Desc& desc, void* inWindowHandle)
 #endif
     combiner.add(DeviceCheckFlag::UseHardwareDevice, ChangeType::OnOff);        ///< First try hardware, then reference
     
-    ComPtr<IDXGIFactory4> dxgiFactory;
-    ComPtr<IDXGIAdapter> adapter;
+    const D3D_FEATURE_LEVEL featureLevel = D3D_FEATURE_LEVEL_11_0;
+
     const int numCombinations = combiner.getNumCombinations();
     for (int i = 0; i < numCombinations; ++i)
     {
-        adapter.setNull();
-        if (SLANG_SUCCEEDED(_createAdaptor(combiner.getCombination(i), dxgiFactory, adapter)))
+        if (SLANG_SUCCEEDED(_createAdaptor(combiner.getCombination(i), featureLevel, m_adapterInfo)))
         {
             break;
         }
     }
 
-    if (!adapter)
+    if (!m_adapterInfo.m_adapter)
     {
         // Couldn't find an adapter
         return SLANG_FAIL;
     }
 
+    // Set the device
+    m_device = m_adapterInfo.m_device;
+
     // Find what features are supported
     {
         // Check this is how this is laid out...
         SLANG_COMPILE_TIME_ASSERT(D3D_SHADER_MODEL_6_0 == 0x60);
 
-        D3D12_FEATURE_DATA_SHADER_MODEL featureShaderMode;
-        featureShaderMode.HighestShaderModel = D3D_SHADER_MODEL(0x62);
+        {
+            D3D12_FEATURE_DATA_SHADER_MODEL featureShaderModel;
+            featureShaderModel.HighestShaderModel = D3D_SHADER_MODEL(0x62);
 
-        if (SLANG_SUCCEEDED(m_device->CheckFeatureSupport(D3D12_FEATURE_SHADER_MODEL, &featureShaderMode, sizeof(featureShaderMode))) &&
-            featureShaderMode.HighestShaderModel >= 0x62)
+            // TODO: Currently warp causes a crash when using half, so disable for now
+            if (SLANG_SUCCEEDED(m_device->CheckFeatureSupport(D3D12_FEATURE_SHADER_MODEL, &featureShaderModel, sizeof(featureShaderModel))) &&
```

Comment:
Style nit: having `&&` on the end of the (long) line makes it harder to pick out the structure of a compound condition like this. That's the basic reason why I personally prefer to wrap binary operators like this:

```
if(aVeryLongConditionThatTakesUpMostOfALine
    && b
    && c)
{ ... }
```

over this:

```
if(aVeryLongConditionThatTakesUpMostOfALine &&
    b &&
    c)
{ ... }
```